framework/include/actioncomponents/ComponentBoundaryConditionInterface.h

```diff
@@ -0,0 +1,98 @@
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "ActionComponent.h"
+#include "InputParameters.h"
+#include "MooseTypes.h"
+
+/**
+ * Helper class to help Components accept boundary condition parameters that the Physics may use
+ * to generate the adequate boundary conditions
+ * Note: Trying out virtual inheritance. It makes things
+ *       a little easier to define as we can use the attributes
+ *       of the underlying ActionComponent
+ */
+class ComponentBoundaryConditionInterface : public virtual ActionComponent
+{
+public:
+  static InputParameters validParams();
+
+  ComponentBoundaryConditionInterface(const InputParameters & params);
+
+  enum BoundaryConditionType
+  {
+    FIXED_VALUE,
+    FLUX
+  };
+
+  /**
+   * Whether the component has a boundary condition parameter is specified for the requested
+   * variable
+   * @param variable the name of the variable
+   * @return Whether the component has a boundary condition parameter specified for the requested
+   * variable
+   */
+  bool hasBoundaryCondition(const VariableName & variable) const;
+  /**
+   * Whether the component has a boundary condition parameter is set for the requested variable and
+   * boundary
+   * @param variable the name of the variable
+   * @param boundary the name of the boundary
+   * @return Whether the component has a boundary condition parameter specified
+   */
+  bool hasBoundaryCondition(const VariableName & variable, const BoundaryName & boundary) const;
+  /**
+   * Get the name of the boundaries on which the variable should have a boundary condition
+   * @param variable the name of the variable concerned
+   * @param requestor name of the requestor for the boundary condition, used in a potential error
+   * message
+   * @return the name of the boundaries
+   */
+  std::vector<BoundaryName> getBoundaryConditionsBoundaries(const VariableName & variable) const;
+  /**
+   * Get the name of the functor providing the boundary condition for the requested variable and
+   * boundary
+   * @param variable the name of the variable concerned
+   * @param boundary the name of the boundary concerned
+   * @param requestor name of the requestor for the boundary condition, used in a potential error
+   * message
+   * @param bc_type the type of the boundary condition (flux or fixed value)
+   * @return the name of the functor providing the value of the boundary condition
+   */
+  const MooseFunctorName getBoundaryCondition(const VariableName & variable,
+                                              const BoundaryName & boundary,
+                                              const std::string & requestor_name,
+                                              BoundaryConditionType & bc_type) const;
```

Comment:
Is this changed in the method?

Reply:
yup this is effectively a return

want both returns to be argu,ments?

Reply:
It doesn't really matter to me, just as long as it's clear that `bc_type` is an "output" in the doxygen